scheduler_perf: "nil pointer dereference" detected with the metrics other than "Histogram"

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
scheduler_perf now is not able to support metrics other than "Histogram", if other type of "metrics" is configured in test, it will throw below exception, 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1369a5b]
```

**What you expected to happen**:
`scheduler_perf` should show helpful error message instead of `panic: runtime error`.
  

**How to reproduce it (as minimally and precisely as possible)**:
Define testcase with `MetricsCollectorConfig` in the config which includes a `SummaryVec`, `scheduler_scheduling_duration_seconds` for example.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

--- a/test/integration/scheduler_perf/util.go
+++ b/test/integration/scheduler_perf/util.go
@@ -179,7 +179,10 @@ func collectHistogram(metric string, labels map[string]string) *DataItem {
 		klog.Error(err)
 		return nil
 	}
-
+	if hist.Histogram == nil {
+		klog.Errorf("metric %q is not a Histogram metric", metric)
+		return nil
+	}
 	if err := hist.Validate(); err != nil {
 		klog.Error(err)
 		return nil

